lib/cli/command-processor.js

```diff
@@ -88,31 +88,44 @@ module.exports = class CommandProcessor extends EventEmitter {
   }
 
   call(interfaceName, methodName, args, callback) {
-    if (!this.cli.connectInitiated) {
-      callback(new Error('Not connected'));
-      return;
-    }
     if (!this.cli.connection) {
-      callback(new Error('Connection in progress'));
+      callback(new Error('Not connected'));
       return;
     }
     this.cli.connection.callMethod(interfaceName, methodName, args, callback);
   }
 
   event(interfaceName, eventName, args, callback) {
-    if (!this.cli.connectInitiated) {
+    if (!this.cli.connection) {
       callback(new Error('Not connected'));
       return;
     }
+    this.cli.connection.emitRemoteEvent(interfaceName, eventName, args);
+    callback();
+  }
+
+  connect(protocol, host, port, app, interfaces, callback) {
+    this.disconnect(() => {
```

Comment:
Does this work? (I expect you to get `Not connected` error when you manually connect for the first time)

Reply:
@lundibundi, error is ignored in the callback for the `disconnect` method here.

Although, I think it's better to make some kind of internal `_disconnect` method that doesn't check if the connection was established before and call it inside of both `connect` and `disconnect` methods. That way you can also avoid additional `_connect` method creation.

Reply:
Yet one question stands: should we close a connection if the initial connection failed or should we let it reconnect?

I say we close it because this is probably because of an error in address, application name or method name.
@belochub @lundibundi what do you think?